tests/unittest_byte.c

```diff
@@ -0,0 +1,62 @@
+#include <check.h>
+
+#include "byte.h"
+
+START_TEST(test_byte_equal)
+{
+  ck_assert_int_ne(byte_equal("", 0, ""), 0);
```

Comment:
I might prefer `ck_assert_int_eq(..., 1)` if we were keeping these tests around. I’m really happy they’re here for just long enough to assure the transition to libc calls. Thank you for writing them.

Reply:
The return value is only used to be checked for being ```if (byte_equal(…))``` or ```if (!byte_equal(…))```, so it's only important if it is 0 or not. In fact, the return value is the difference between the 2 characters in the string (e.g. ```'o' - 'e'```), but we don't care. And we can't and need not to make sure that this is the same return value in the libc implementation.

Reply:
Aha! Thank you for this explanation.